### PR TITLE
Client refactor

### PR DESCRIPTION
**Pull request by Frando**

### CODE REVIEW

**Comment by Frando** [comment]
another update: removes axios and uses [isomorphic-fetch](https://github.com/matthew-andrews/isomorphic-fetch) instead. `Client.fetch` is a wrapper with some extra options, notably `requestType`. Aim would be to keep this low. This could also be where we accept sonar specific url handlers. Much better than before already this way I think. `collection.fetch` just prepends the collection name to the path.

**Comment by Frando** [comment]
I think this is ready for review and adding JSdoc comments for everything that feels good already. We can leave out the legacy client for now and only document the new public classes and methods.

File: sonar-client/lib/collection.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+  async put (record) {
+    return this.fetch('db', {
+      method: 'PUT',
+      body: record
+    })
+  }
+
+  async get (req, opts) {
+    // TODO: Implement RecordCache.has
```
Comment (hng) [inline sonar-client/lib/collection.js:65]: Create issue?

Comment (hng) [reply sonar-client/lib/collection.js:65]: Also cache is already implemented for query? Why not here?

File: sonar-client/lib/legacy-client.js [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    if (typeof args === 'string') {
+      args = JSON.stringify(args)
+    } else if (args instanceof SearchQueryBuilder) {
+      args = args.getQuery()
+    }
+    const collection = await this.focusedCollection()
+    return collection.query('search', args, opts)
+  }
+
+  // TODO: Port!
```
Comment (hng) [inline sonar-client/lib/legacy-client.js:87]: What needs to be done? Issue?

File: sonar-client/index.js [hunk 3]
```diff
@@ -1,8 +1,15 @@
...
+// TODO:
+// Currently this exports the LegacyClient by default. Code can switch to the new client
+// by importing { Client } by default.
 module.exports = function (...args) {
-  return new SonarClient(...args)
+  return new LegacyClient(...args)
 }
-module.exports.SonarClient = SonarClient
+module.exports.SonarClient = LegacyClient
+// TODO: Rethink this, or find better name.
```
Comment (hng) [inline sonar-client/index.js:12]: Create issue concerning future of QueryBuilder?

File: sonar-client/lib/legacy-client.js (hunk 2 above)
Comment (Frando) [reply sonar-client/lib/legacy-client.js:87]: Leftover, I ported all of them

**Comment by Frando** [comment]
OK, going ahead and merging this!

**Pull request merged**